mcpscanner/core/static_analysis/native_analyzer.py

```
@@ -3608,6 +3654,215 @@ def _py_detect_security_ops(self, node: ast.AST) -> Dict[str, bool]:
+
+        if not reachable:
+            return
+
+        dangerous_refs: Set[str] = set()
+        reachable_names = list(ctx.reachable_functions or [])
+        for resolved_key, target in reachable:
+            if resolved_key not in reachable_names:
+                reachable_names.append(resolved_key)
+            try:
+                ops = self._py_detect_security_ops(target)
+            except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)